### ISSUE TITLE
Contect duplication ignores menuindex

### ISSUE THREAD

**Issue by CyntaxDotDev** [modxbot/migrate#8086]
ReSpawN created Redmine issue ID 8086

This "bug" has been around for quite some time, thought I'd report it.

When duplicating a context, menuindex is completely ignored and as the duplicate resource states, it generates a new menuindex. While that's fine totally on it's own, like working in the same context, it is not ideal for creating a duplicate context where you'd want the same order.

In some way, this is the wrong-do-er. 
modx/model/modx/modresource.class.php @ #989

<pre>        /* set new menuindex */
        $childrenCount = $this->xpdo->getCount('modResource',array('parent' => $this->get('parent')));
        $newResource->set('menuindex',$childrenCount);</pre>

Again, totally fine, but duplicating a context should simply represent a duplicate of the duplicated resource.

A couple of ways to achieve this is simply fetch all resources in the duplicating context, mark them dirty and new and re-save them with the new context_key. Haven't tried it, should (at least partially) work. Another way is fetching all resources and creating new objects on-the-fly in the loop. 
Another way even, is passing along into the array the current menuindex which needs to be respected by the duplicator.

processors/context/duplicate.class.php @ #108

<pre>        $resources = $this->modx->getCollection('modResource',array(
            'context_key' => $this->object->get('key'),
            'parent' => 0,
        ));
        if (count($resources) > 0) {
            /** @var modResource $resource */
            foreach ($resources as $resource) {
                $resource->duplicate(array(
                    'prefixDuplicate' => false,
                    'duplicateChildren' => true,
                    'overrides' => array(
                        'context_key' => $this->newObject->get('key'),
                    ),
                ));
            }
        }</pre>

Simply put, it's very desirable to get this fixed. ;)
